cardano-db-sync/src/Cardano/DbSync/Default.hs

```diff
@@ -213,8 +218,9 @@ insertBlock syncEnv cblk applyRes firstAfterRollback tookSnapshot = do
       when withinHalfHour $ do
         ranIndexes <- liftIO $ getRanIndexes syncEnv
         unless ranIndexes $ do
+          addConstraintsIfNotExist syncEnv
           unless commited DB.transactionCommit
-          liftIO $ runIndexMigrations syncEnv
+    -- liftIO $ runIndexMigrations syncEnv
```

Comment:
@kderme is this what you had in mind?

Reply:
Not exactly. This is different from the indexes creation. `addConstraintsIfNotExist` can  be called after the indexes are created and shouldn't be affected by `ranIndexes` flag.

Reply:
For context: Indexes are automatically created by db-sync when it reaches 30min before the tip of the chain. This allows users to run fast queries. We can also create the unique keys at the same point if they are missing. They are not yet needed by db-sync, but unique keys create automatically indexes, so they can speed up queries by users.

Reply:
ah ok the previous comment stated "(above this unless)" did you mean before `unless ranIndexes $ do`?

Reply:
Yes, it can go above or below (not inside)